packages/core/web-test-runner.config.mjs

```diff
@@ -0,0 +1,49 @@
+/**
+ * Web Test Runner
+ * 
+ * This configures Core unit tests to run using @web/test-runner 
+ */
+
+import { playwrightLauncher } from '@web/test-runner-playwright';
+import { esbuildPlugin } from '@web/dev-server-esbuild';
+import { fromRollup } from '@web/dev-server-rollup';
+import execute from 'rollup-plugin-shell';
+import baseConfig from './web-dev-server.config.mjs';
+
+export default /** @type {import("@web/test-runner").TestRunnerConfig} */ ({
+  // uncomment open/manual to debug in browser
+  // open: true,
+  // manual: true,
+  files: ['./src/**/*.spec.ts'],
+  testsFinishTimeout: 20000,
+  browsers: [playwrightLauncher({ product: 'chromium' })],
+  coverageConfig: {
+    require: ['ts-node/register'],
+    extension: ['.ts'],
+    exclude: ['**/*.d.ts', '**/*.scss.js', '**/node_modules/**', '**/test/**', '**/dist/core/**/index.js', '**/dist/core/**/register.js'],
+    report: true,
+    reportDir: 'dist/coverage',
+    threshold: {
+      statements: 90,
+      branches: 85,
+      functions: 85,
+      lines: 90,
+    },
+  },
+  nodeResolve: true,
+  plugins: [
+    ...baseConfig.plugins,
+    esbuildPlugin({ ts: true, json: true, target: 'auto' }),
+    fromRollup(execute)({ commands: [`tsc --noEmit src/**/*.spec.ts`], hook: 'writeBundle' })
```

Comment:
Runs the tests against the rollup build output in the dist but will compile the spec TS files on the fly. This ensures the tests run against the same code that is published to npm.